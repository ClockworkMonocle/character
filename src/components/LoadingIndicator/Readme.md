For when you want to load stuff and be indeterminate about it.

```js
<LoadingIndicator />
```
