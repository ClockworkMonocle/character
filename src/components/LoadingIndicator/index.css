
.loading {
  display: flex;
  flex-direction: row;
}

.loading_dot {
  width: 12px;
  height: 12px;
  border-radius: 50%;

  background-color: var(--attack-color);
  animation-name: loadingAnim;
  animation-duration: 500ms;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

.loading_dot:nth-of-type(2) {
  animation-delay: 150ms;
  margin-left: 4px;
  margin-right: 4px;
}

.loading_dot:nth-of-type(3) {
  animation-delay: 300ms;
}

@keyframes loadingAnim {
  from {
    opacity: 0;
    transform: scale(0);
  }

  to {
    opacity: 1;
    transform: scale(1);
  }
}
