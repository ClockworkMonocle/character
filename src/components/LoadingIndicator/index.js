import React from 'react';

import './index.css';

export default function LoadingIndicator() {
  return (
    <div className="loading">
      <div className="loading_dot" />
      <div className="loading_dot" />
      <div className="loading_dot" />
    </div>
  );
}
