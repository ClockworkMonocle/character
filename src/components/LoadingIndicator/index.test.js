import React from 'react';
import { shallow } from 'enzyme';
import LoadingIndicator from './index';

test('<LoadingIndicator /> renders', () => {
  expect(shallow(<LoadingIndicator />)).toMatchSnapshot();
});
