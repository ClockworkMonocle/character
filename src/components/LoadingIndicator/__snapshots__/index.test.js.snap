// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> renders 1`] = `
<div
  className="loading"
>
  <div
    className="loading_dot"
  />
  <div
    className="loading_dot"
  />
  <div
    className="loading_dot"
  />
</div>
`;
